runner/runners/statuses.go

```diff
@@ -0,0 +1,169 @@
+package runners
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/scootdev/scoot/runner"
+)
+
+const UnknownRunIDMsg = "unknown run id %v"
+
+// NewStatuses creates a new empty Statuses
+func NewStatuses() *Statuses {
+	return &Statuses{runs: make(map[runner.RunID]runner.RunStatus)}
+}
+
+// Statuses is a database of RunStatus'es. It allows clients to Write Statuses, Query the
+// current status, and listen for updates to status. It implements runner.RunStatus
+type Statuses struct {
+	mu        sync.Mutex
+	runs      map[runner.RunID]runner.RunStatus
+	nextRunID int64
+	listeners []queryAndCh
+}
+
+type queryAndCh struct {
+	q  runner.Query
+	ch chan runner.RunStatus
+}
+
+// Writer interface
+
+// NewRun creates a new RunID in state Preparing
+func (s *Statuses) NewRun() (runner.RunStatus, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	id := runner.RunID(strconv.FormatInt(s.nextRunID, 10))
+	s.nextRunID++
+
+	st := runner.RunStatus{
```

Comment:
Commented in task_runner.go that we need to reenable this.

(As discussed with Jason off-line, turning this off for now to make this PR easier to land)